--- a/libgo/go/net/textproto/reader.go
+++ b/libgo/go/net/textproto/reader.go
@@ -498,7 +498,7 @@ func (r *Reader) ReadMIMEHeader() (MIMEH
 			// Most headers aren't multi-valued.
 			// Set the capacity on strs[0] to 1, so any future append
 			// won't extend the slice into the other strings.
-			vv, strs = strs[:1:1], strs[1:]
+			vv, strs = strs[0:1], strs[1:]
 			vv[0] = value
 			m[key] = vv
 		} else {
