--- a/libgo/runtime/getncpu-linux.c
+++ b/libgo/runtime/getncpu-linux.c
@@ -7,6 +7,9 @@
 
 // CPU_COUNT is only provided by glibc 2.6 or higher
 #if !defined(__GLIBC_PREREQ) || !__GLIBC_PREREQ(2, 6)
+#ifdef CPU_COUNT
+#undef CPU_COUNT
+#endif
 #define CPU_COUNT(set) _CPU_COUNT((unsigned int *)(set), sizeof(*(set))/sizeof(unsigned int))
 static int _CPU_COUNT(unsigned int *set, size_t len) {
 	int cnt;
